### PR TITLE
Fix integration tests

### PR DESCRIPTION
**Pull request by bhaskaramVaranasi**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,6 +7,7 @@ require (
 	github.com/davecgh/go-spew v1.1.1
 	github.com/ghodss/yaml v1.0.0
 	github.com/gogo/protobuf v1.3.1
+	github.com/kr/pty v1.1.5 // indirect
```
Comment (RuiGuo34) [inline go.mod:10]: possible to remove indirect package?

File: test/integration/authz_v1_test.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ import (
 
 	"istio.io/api/rbac/v1alpha1"
 	"istio.io/istio/pilot/pkg/model"
-	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (RuiGuo34) [inline test/integration/authz_v1_test.go:17]: doesn't need to specify v1 in the beginning as package name is v1

File: test/integration/authz_v1_test.go [hunk 3]
```diff
@@ -119,9 +120,6 @@ func TestCreateServiceRoleOnly(t *testing.T) {
 	}
 	r := fixtures.GetExpectedRbac(o)
 	rolloutAndValidateRbac(t, r, create)
-	namespace := athenz.DomainToNamespace(r.AD.Name)
```
Comment (RuiGuo34) [inline test/integration/authz_v1_test.go:122]: Why is this part removed?

File: test/integration/fixtures/fixtures.go [hunk 4]
```diff
@@ -6,18 +6,20 @@ package fixtures
...
 	"github.com/ardielle/ardielle-go/rdl"
 	"github.com/yahoo/athenz/clients/go/zms"
 	"github.com/yahoo/k8s-athenz-istio-auth/pkg/athenz"
 	"github.com/yahoo/k8s-athenz-istio-auth/pkg/istio/rbac/common"
-	athenzdomain "github.com/yahoo/k8s-athenz-syncer/pkg/apis/athenz/v1"
-
 	"istio.io/api/rbac/v1alpha1"
 	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/config/schema/collections"
+	v1 "k8s.io/api/core/v1"
```
Comment (RuiGuo34) [inline test/integration/fixtures/fixtures.go:17]: no need to mentioned v1 again

File: test/integration/fixtures/fixtures.go [hunk 5]
```diff
@@ -6,18 +6,20 @@ package fixtures
...
 
-	"k8s.io/api/core/v1"
+	"strings"
+
+	athenzdomain "github.com/yahoo/k8s-athenz-syncer/pkg/apis/athenz/v1"
+	_ "istio.io/istio/pkg/config/schema/collections"
```
Comment (RuiGuo34) [inline test/integration/fixtures/fixtures.go:22]: what is the use of _ here?

File: test/integration/go.mod [hunk 6]
```diff
@@ -3,51 +3,59 @@ module github.com/yahoo/k8s-athenz-istio-auth/test/integration
...
+	github.com/yahoo/k8s-athenz-syncer v0.1.7
+	go.etcd.io/etcd v3.3.25+incompatible
+	istio.io/api v0.0.0-20200513175333-ae3da0d240e3
+	istio.io/client-go v0.0.0-20200513180646-f8d9d8ff84e6
+	istio.io/istio v0.0.0-20200708165503-80f49905d910
+	istio.io/pkg v0.0.0-20200324191837-25e6bb9cf135
+	k8s.io/api v0.17.14
+	k8s.io/apiextensions-apiserver v0.17.14
+	k8s.io/apimachinery v0.17.14
 	k8s.io/client-go v11.0.0+incompatible
```
Comment (RuiGuo34) [inline test/integration/go.mod:29]: can we upgrade client-go version as well?

File: test/integration/go.mod [hunk 7]
```diff
@@ -3,51 +3,59 @@ module github.com/yahoo/k8s-athenz-istio-auth/test/integration
...
-	k8s.io/kubernetes v1.13.1
+	k8s.io/kubernetes v1.20.5
```
Comment (RuiGuo34) [inline test/integration/go.mod:30]: can we keep same version as v0.17.14

File: test/integration/authz_v1_test.go (hunk 3 above)
Comment (kpharasi) [reply test/integration/authz_v1_test.go:122]: +1

File: test/integration/go.mod (hunk 7 above)
Comment (bhaskaramVaranasi) [reply test/integration/go.mod]: Done

File: test/integration/go.mod (hunk 6 above)
Comment (bhaskaramVaranasi) [reply test/integration/go.mod]: Done

File: test/integration/fixtures/fixtures.go (hunk 5 above)
Comment (bhaskaramVaranasi) [reply test/integration/fixtures/fixtures.go]: Unwanted removed

File: test/integration/fixtures/fixtures.go (hunk 4 above)
Comment (bhaskaramVaranasi) [reply test/integration/fixtures/fixtures.go]: Removed v1

File: go.mod (hunk 1 above)
Comment (bhaskaramVaranasi) [reply go.mod]: Removed this.

File: test/integration/authz_v1_test.go (hunk 2 above)
Comment (bhaskaramVaranasi) [reply test/integration/authz_v1_test.go]: Okay, removed it.

File: test/integration/authz_v1_test.go (hunk 3 above)
Comment (bhaskaramVaranasi) [reply test/integration/authz_v1_test.go:122]: In this case we still end up creating a ServiceRoleBinding containing the  [Spiffe URI](https://github.com/yahoo/k8s-athenz-istio-auth/blob/464d0c40214dfb289bc1b8f0eaa079701baf6235/test/integration/fixtures/fixtures.go#L250-L252), so the test case is no longer valid.

File: test/integration/authz_v1_test.go [hunk 8]
```diff
@@ -110,7 +111,7 @@ func TestCreateServiceRoleAndBinding(t *testing.T) {
 }
 
 // 1.1 Create SR only if there are no role members for SRB creation
```
Comment (mdhume) [inline test/integration/authz_v1_test.go:113]: let's update the comment - `create SRB with role cert spiffe only if there are no role members`

File: test/integration/fixtures/fixtures.go [hunk 9]
```diff
@@ -239,14 +240,19 @@ func GetExpectedRbac(o *OverrideRbac) *ExpectedRbac {
...
-		if len(signedDomain.Domain.Roles[i].RoleMembers) > 0 {
-			srbSpec.RoleRef.Name = sr.Name
-			srb := common.NewConfig(model.ServiceRoleBinding.Type, ns, roleName, srbSpec)
-			modelConfig = append(modelConfig, srb)
+		// Every Role has SRB with Spiffe URI
+		if len(signedDomain.Domain.Roles[i].RoleMembers) == 0 {
+			srbSpec.Subjects = []*v1alpha1.Subject{}
 		}
+		srbSpec.Subjects = append(srbSpec.Subjects, &v1alpha1.Subject{
+			User: fmt.Sprintf("%s/ra/%s", signedDomain.Domain.Name, sr.Name),
```
Comment (mdhume) [inline test/integration/fixtures/fixtures.go:251]: we should ideally also put this in `getDefaultServiceRoleBinding` since all SRB should have role cert spiffe added by default now

Comment (bhaskaramVaranasi) [reply test/integration/fixtures/fixtures.go:251]: `getDefaultServiceRoleBinding` does not have any namespace info so moved adding spiffe uri here where namespace info is available.

File: test/integration/authz_v1_test.go (hunk 8 above)
Comment (bhaskaramVaranasi) [reply test/integration/authz_v1_test.go]: Updated the comment

File: test/integration/authz_v1_test.go [hunk 10]
```diff
@@ -110,7 +111,7 @@ func TestCreateServiceRoleAndBinding(t *testing.T) {
...
-func TestCreateServiceRoleOnly(t *testing.T) {
+func TestCreateServiceRoleAndBindingsWhenNoMemebersInRole(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (bhaskaramVaranasi) [reply test/integration/authz_v1_test.go]: Fixed

**Review by kpharasi — APPROVED** [review]
👍

**Review by RuiGuo34 — APPROVED** [review]

**Comment by mcieplak** [comment]
@bhaskaramVaranasi Can we also make sure a failure in the integration tests causes the travis ci build to fail?

**Review by mdhume — APPROVED** [review]

**Comment by bhaskaramVaranasi** [comment]
> @bhaskaramVaranasi Can we also make sure a failure in the integration tests causes the travis ci build to fail?

I will take this up in the next PR.
